controllers/progressivesync_controller.go

```diff
@@ -418,12 +431,44 @@ func (r *ProgressiveSyncReconciler) syncApp(ctx context.Context, appName string)
 	return r.ArgoCDAppClient.Sync(ctx, &syncReq)
 }
 
-// patchStatus patches the progressive sync object status
-func (r *ProgressiveSyncReconciler) patchStatus(ctx context.Context, pr *syncv1alpha1.ProgressiveSync) error {
-	key := client.ObjectKeyFromObject(pr)
-	latest := syncv1alpha1.ProgressiveSync{}
-	if err := r.Client.Get(ctx, key, &latest); err != nil {
-		return err
-	}
-	return r.Client.Status().Patch(ctx, pr, client.MergeFrom(&latest))
+// setStatus updates the progressive sync object status with backoff
+func (r *ProgressiveSyncReconciler) setStatus(ctx context.Context, pr *syncv1alpha1.ProgressiveSync) error {
+
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+		key := client.ObjectKeyFromObject(pr)
+		latest := syncv1alpha1.ProgressiveSync{}
+		if err := r.Client.Get(ctx, key, &latest); err != nil {
+			return err
+		}
+		latest.Status = pr.Status
+		if err := r.Client.Status().Update(ctx, &latest); err != nil {
+			return err
+		}
+		pr = &latest
+		return nil
+
+	})
+	return retryErr
+}
+
+// initializeStatus initializes the progressive sync object status to prevent conflicts when updating status later on
+func (r *ProgressiveSyncReconciler) initializeStatus(ctx context.Context, pr *syncv1alpha1.ProgressiveSync) error {
```

Comment:
semantics, but perhaps `initializeStatus` should be `resetStatus`? and the outcome of `reset` would be `initialized`?